scripts/pre_commit_hook.py

```diff
@@ -106,6 +120,58 @@ def _does_current_folder_contain_have_package_lock_file():
     return os.path.isfile('package-lock.json')
 
 
+def check_changes(filetype):
+    """Checks if diff in feconf or constants file includes
+    changes made for release.
+
+    Args:
+        filetype: str. The file to check - feconf or constants.
+
+    Returns:
+        bool. Whether the diff includes changes made for release.
+    """
+    if filetype == 'feconf':
+        filepath = FECONF_FILEPATH
+        keys_to_check = KEYS_UPDATED_IN_FECONF
+        assignment_op = '= '
+    else:
+        filepath = CONSTANTS_FILEPATH
+        keys_to_check = KEYS_UPDATED_IN_CONSTANTS
+        assignment_op = ': '
+
+    diff_output = subprocess.check_output([
+        'git', 'diff', filepath])[:-1].split('\n')
+    for index, line in enumerate(diff_output):
+        if index + 1 >= len(diff_output):
+            continue
+        next_line = diff_output[index + 1]
+        if line.startswith('-') and next_line.startswith('+') and any(
+                key in line for key in keys_to_check):
```

Comment:
Lucky that no string is a substring of another in keys_to_check...

Reply:
Yes :P

Reply:
Should we add a comment above saying this might break if one is a substring of the other?

Reply:
Should I check instead that the line contains `key = ` or `"key": `?

Reply:
Updated according to this (There would be no issue now even if a key is a substring of other key!)

Reply:
maybe yeah. It is a smallish array anyway.

Reply:
Sounds good.